src/main/java/org/janelia/saalfeldlab/n5/KeyValueAccess.java

```diff
@@ -80,13 +80,28 @@ public default String[] components(final String path) {
 	 */
 	public default String compose(final URI uri, final String... components) {
 
-		if (components.length == 0)
+		int firstNonEmptyIdx = 0;
+		while (firstNonEmptyIdx < components.length && components[firstNonEmptyIdx].isEmpty()) {
+			firstNonEmptyIdx++;
+		}
+
+		/*If there are no non-empty components, there is nothing to compose against; return the uri. */
+		if (components.length == firstNonEmptyIdx)
 			return uri.toString();
 
-		/* add the initial path to the components */
-		final String[] allComponents = new String[components.length + 1];
-		allComponents[0] = uri.getPath();
-		System.arraycopy(components, 0, allComponents, 1, components.length);
+		/* allocate space for the initial path and the new components, skipping empty strings  */
+		final int nonEmptysize = components.length - firstNonEmptyIdx;
+		final String[] allComponents = new String[1 + nonEmptysize];
+		if (uri.getPath().isEmpty())
```

Comment:
This section adds a new component `"/"` at the start when the uri path is empty to deal with the behavior you described [here](https://github.com/saalfeldlab/n5/pull/152#issue-3050701409):

```java
URI.create("http://localhost:8000").resolve("bar")  // -> "http://localhost:8000bar"
```

Will this do what we expect if components[0] has a leading slash too?